drivers/usb_c/ppc/usbc_ppc_numaker.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_ppc
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/usb_c/usbc_ppc.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_numaker.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ppc_numaker, CONFIG_USBC_LOG_LEVEL);
+
+#include <soc.h>
+#include <NuMicro.h>
+
+/* Implementation notes on NuMaker TCPC/PPC/VBUS
+ *
+ * PPC and VBUS rely on TCPC/UTCPD and are just pseudo. They are completely
+ * implemented in TCPC/UTCPD.
+ */
+
+/* Immutable device context */
+struct numaker_ppc_config {
+	const struct device *tcpc_dev;
+};
+
+/* Mutable device context */
+struct numaker_ppc_data {
+	uint32_t dummy;
+};
+
+/* TCPC exported */
+int numaker_tcpc_ppc_is_dead_battery_mode(const struct device *dev);
+int numaker_tcpc_ppc_exit_dead_battery_mode(const struct device *dev);
+int numaker_tcpc_ppc_is_vbus_source(const struct device *dev);
+int numaker_tcpc_ppc_is_vbus_sink(const struct device *dev);
+int numaker_tcpc_ppc_set_snk_ctrl(const struct device *dev, bool enable);
+int numaker_tcpc_ppc_set_src_ctrl(const struct device *dev, bool enable);
+int numaker_tcpc_ppc_set_vbus_discharge(const struct device *dev, bool enable);
+int numaker_tcpc_ppc_is_vbus_present(const struct device *dev);
+int numaker_tcpc_ppc_set_event_handler(const struct device *dev, usbc_ppc_event_cb_t handler,
+				       void *data);
+int numaker_tcpc_ppc_dump_regs(const struct device *dev);
+
+static int numaker_ppc_init(const struct device *dev)
+{
+	const struct numaker_ppc_config *const config = dev->config;
+	const struct device *tcpc_dev = config->tcpc_dev;
+	int rc = 0;
+
+	/* Rely on TCPC */
+	if (!device_is_ready(tcpc_dev)) {
+		LOG_ERR("TCPC device not ready");
+		rc = -ENODEV;
```

Comment:
You can just ```return -ENODEV``` here for failure, and ```return 0``` below for success.

Reply:
fixed